packages/snjs/lib/protocol/collection/notes_display_criteria.ts

```diff
@@ -90,7 +90,15 @@ export class NotesDisplayCriteria {
       }
     }
     if (this.searchQuery) {
-      filters.push((note) => noteMatchesQuery(note, this.searchQuery!));
+      filters.push((note) => {
+        const noteIsAMatch = noteMatchesQuery(note, this.searchQuery!);
+
+        const noteTags = collection.elementsReferencingElement(note)
+          .filter((item) => item.content_type === ContentType.Tag) as SNTag[];
```

Comment:
I filtered through the result of `elementsReferencingElement` to make sure all items are of type `SNTag`.

Can a note have references to other items that are not `SNTag`? If so, [this line](https://github.com/standardnotes/snjs/blob/210ba3a0879cb483865469fbbcb3276fad6fa44a/packages/snjs/lib/protocol/collection/notes_display_criteria.ts#L80) should have a safe check.